jc/parsers/xrandr.py

```diff
@@ -294,15 +375,67 @@ def _parse_device(next_lines: List[str], quiet: bool = False) -> Optional[Device
                     [f"{next_line} : {k} - {v} is not int-able"]
                 )
 
+    model: Optional[Model] = _parse_model(next_lines, quiet)
+    if model:
+        device["model_name"] = model["name"]
+        device["product_id"] = model["product_id"]
+        device["serial_number"] = model["serial_number"]
+
     while next_lines:
         next_line = next_lines.pop()
         next_mode: Optional[Mode] = _parse_mode(next_line)
         if next_mode:
             device["associated_modes"].append(next_mode)
         else:
+            if re.match(_device_pattern, next_line):
+                next_lines.append(next_line)
+                break
+    return device
+
+
+# EDID:
+#      00ffffffffffff004ca3523100000000
+#      0014010380221378eac8959e57549226
+#      0f505400000001010101010101010101
+#      010101010101381d56d4500016303020
+#      250058c2100000190000000f00000000
+#      000000000025d9066a00000000fe0053
+#      414d53554e470a204ca34154000000fe
+#      004c544e313536415432343430310018
+_edid_head_pattern = r"\s*EDID:\s*"
+_edid_line_pattern = r"\s*(?P<edid_line>[0-9a-fA-F]{32})\s*"
+
+
+def _parse_model(next_lines: List[str], quiet: bool = False) -> Optional[Model]:
```

Comment:
It looks like the type annotation here (specifically referring to `Model`) is breaking python versions 3.6 and 3.7. Not sure why, but maybe we just modify/remove the type annotation here and see if that will allow the tests to pass? Type annotations seem to be a bit broken in this parser, overall.

Reply:
Sure, we can do that.

The only files the xrandr parser needs, in order to decode the EDID data, is the Edid and EdidHelper classes. I think we can remove any other file entirely and keep those classes having them modified a bit by removing any @staticmethod, @classmethod annotations.

Let me know if there is a way to run the tests which are failing for centos to save some from back and forth time. I suppose the `runtests.sh` script does not include those tests, right? I ran this script and all test are passing.

Reply:
All the tests run with `./runtests.sh`, but github actions just runs the same tests several times with different Python versions and operating systems. The way to do it on your local machine is by using something like [pyenv](https://github.com/pyenv/pyenv) so you can change to an older Python version and run `./runtests.sh` again.